variantlib/models/validators.py

```diff
@@ -32,21 +33,25 @@ def validate_list_min_len(values: list, min_length: int) -> None:
         )
 
 
-def validate_list_all_unique(values: list[Any], key: None | Callable = None) -> None:
+def validate_list_all_unique(values: list[Any], keys: list[str] | None = None) -> None:
     """
     Validate that all elements in the list are unique.
     Raises a ValueError if duplicates are found.
     """
     seen = set()
 
     for value in values:
-        if key is not None:
-            value = key(value)  # noqa: PLW2901
+        _value = value
 
-        if value in seen:
-            raise ValidationError(f"Duplicate value found: '{value}' in list.")
+        if keys is not None:
+            _value = tuple([attrgetter(key)(value) for key in keys])
```

Comment:
```suggestion
            _value = tuple([getattr(value, key) for key in keys])
```

`attrgetter()` only makes sense when you don't have the value immediately.